### DISCUSSION TITLE
Add a Claims Transformer

### DISCUSSION THREAD

**Discussion by Stonebkfly** [oidc-wp/openid-connect-generic#350] [Q&A] [answered]
The Identity Provider that I am using does not contain all the claims that I would like to supply to the WP site. (I would rather not update the Identity Provider as I want to keep it as simple as possible. ) Where would be a good place to add a Claims Transformer in WP that would construct some new claims before the login/account creation process occurs? This Claims Transformer would need to reach out to an API endpoint with an ID that is provided in the claims to get the other claims that need to be added.
Thank you!

**Comment by timnolte** [comment, marked as answer]
@stonebkfly you should be able to use the following action hook to do your work. https://github.com/oidc-wp/openid-connect-generic/blob/dev/HOWTO.md#openid-connect-generic-update-user-using-current-claim

**Reply by Stonebkfly** [reply]
Was able to achieve what I needed with openid-connect-generic-update-user-using-current-claim and openid-connect-generic-user-create! Thank you!
